src/libs/history.c

```diff
@@ -169,6 +175,67 @@ static GtkWidget *_lib_history_create_button(dt_lib_module_t *self, int num, con
   return widget;
 }
 
+static GList *_duplicate_history(GList *hist)
+{
+  GList *result = NULL;
+
+  GList *h = g_list_first(hist);
+  while(h)
+  {
+    const dt_dev_history_item_t *old = (dt_dev_history_item_t *)(h->data);
+
+    dt_dev_history_item_t *new = (dt_dev_history_item_t *)g_malloc(sizeof(dt_dev_history_item_t));
```

Comment:
And you have got yourself alloc-dealloc-mismatch
Better to just stick to non-glib memory allocation functions everywhere.